src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -551,11 +548,29 @@ ImmutableArray<ISymbolInitializer> IObjectCreationExpression.MemberInitializers
                             }
                             return builder.ToImmutableAndFree();
                         }
+
                         return ImmutableArray<ISymbolInitializer>.Empty;
                     });
             }
         }
 
+        internal static ImmutableArray<BoundExpression> GetChildInitializers(BoundExpression objectOrCollectionInitializer)
+        {
+            var objectInitializerExpression = objectOrCollectionInitializer as BoundObjectInitializerExpression;
+            if (objectInitializerExpression != null)
+            {
+                return objectInitializerExpression.Initializers;
+            }
+
+            var collectionInitializerExpresion = objectOrCollectionInitializer as BoundCollectionInitializerExpression;
+            if (collectionInitializerExpresion != null)
+            {
+                return collectionInitializerExpresion.Initializers;
```

Comment:
Makes sense, let me revert the changes to IObjectCreationExpression.MemberInitializers implementation.